src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj

```diff
@@ -0,0 +1,28 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <IsShipping>false</IsShipping>
+  </PropertyGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\ServiceHub\Microsoft.CodeAnalysis.Remote.ServiceHub.csproj" />
+  </ItemGroup>
+  <ItemGroup>
+    <!-- These references need to be deployed to the vsix subfolder containing servicehub bits for .Net Core -->
+    <PackageReference Include="Microsoft.VisualStudio.Telemetry" Version="$(MicrosoftVisualStudioTelemetryVersion)" />
+    <PackageReference Include="SQLitePCLRaw.bundle_green" Version="$(SQLitePCLRawbundle_greenVersion)" />
+  </ItemGroup>
+
+  <Target Name="PublishProjectOutputGroup" DependsOnTargets="Publish" Returns="@(_PublishedFiles)">
+    <ItemGroup>
+      <!-- Need to include and then update items (https://github.com/microsoft/msbuild/issues/1053) -->
+      <_PublishedFiles Include="$(PublishDir)**\*.*" Exclude="$(PublishDir)**\Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.*"/>
+      <_PublishedFiles Remove="@(_PublishedFiles)" Condition="'%(Extension)' == '.pdb'" />
```

Comment:
@tmat Do you know if there's a way to exclude all system libraries from being included, w/o maintaining an explicit exclusion list here?

Reply:
actually, we need to figure out how to best de-duplicate our own assemblies as well, right now it's a ~66mb increase in our vsix size (i.e. everything this project brings in), almost 35mb of it is Roslyn assemblies that is identical to what we ship for desktop (all targeting netstandard2.0)

Reply:
I don't think we can avoid that. We actually should cross-gen our .NET Core binaries and NGEN our Desktop binaries.

Reply:
The only thing I know of is https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-framework-dependent